pest/src/prec_climber.rs

```diff
@@ -9,13 +9,96 @@
 
 //! Constructs useful in infix operator parsing with the precedence climbing method.
 
-use std::collections::HashMap;
+use std::borrow::Cow;
 use std::iter::Peekable;
 use std::ops::BitOr;
 
 use iterators::Pair;
 use RuleType;
 
+/// Macro for more convenient const fn definition of `prec_climber::PrecClimber`.
+///
+/// # Examples
+///
+/// ```
+/// # use pest::prec_climber::{Assoc, PrecClimber};
+/// # use pest::prec_climber;
+/// # #[allow(non_camel_case_types)]
+/// # #[allow(dead_code)]
+/// # #[derive(Clone, Copy, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+/// # enum Rule {
+/// #     plus,
+/// #     minus,
+/// #     times,
+/// #     divide,
+/// #     power
+/// # }
+/// static CLIMBER: PrecClimber<Rule> = prec_climber![
+///     L   plus | minus,
+///     L   times | divide,
+///     R   power,
+/// ];
+/// ```
+#[cfg(const_prec_climber)]
+#[macro_export]
+macro_rules! prec_climber {
+    ( __assoc L ) => { $crate::prec_climber::Assoc::Left };
```

Comment:
I've been usually using `@foo` for these, but I'm not completely sure there's a precedence in pest for this yet.

Reply:
I've changed it to `@` too, it also allowed me to move the user callable pattern to the top which makes the documentation for the macro a bit more readable.